tests/kafkatest/services/kafka/kafka.py

```diff
@@ -242,8 +243,8 @@ def start_node(self, node):
     def pids(self, node):
         """Return process ids associated with running processes on the given node."""
         try:
-            cmd = "ps ax | grep -i kafka | grep java | grep -v grep | awk '{print $1}'"
-
+            cmd = "jcmd | grep -e org.apache.zookeeper.server.quorum.QuorumPeerMain -e kafka.Kafka" \
```

Comment:
We should probably use jcmd in the kafka shell scripts as well. In a separate PR, of course.